iminuit/_libiminuit.pyx

```diff
@@ -192,10 +195,39 @@ cdef class Minuit:
 
         **Arguments:**
 
-            - **fcn**: the function to be optimized. Minuit automagically finds
-              parameters names. More information about how
-              Minuit detects function signature can be found in
-              :ref:`function-sig-label`
+            **fcn**, the function to be optimized, is the only required argument.
+
+            Two kinds of function signatures are understood.
```

Comment:
@HDembinski  and I had a long discussion on this PR and specifically the questions whether to support a second type of callback function (that takes a sequence) and whether to add a second create classmethod to support the use case currently covered by forced_parameters better.

In the end, it still remains a complex decision, and a matter of preference whether convenience / removed boilerplate for certain use cases is important enough to expand the API.

For me, the bottom line is that my preference would be to mostly keep things as is, concretely:
- add documentation how to do common things with the current API (e.g. using `def f(*pars): pars = np.array(pars)` and explain to users what `*pars` is), not add a second way to do this as proposed here via `def f(pars)` where pars is directly an array
- OK to add a classmethod that helps to init this use case, which basically prepares a `kwargs` dict and calls `init(**kwargs, forced_parameters=True)` but makes this more convenient.

My argument is that there's a hidden cost to adding new features and a second way to do the same thing, because then we need to explain both ways in the docs and users aren't better off really if half of them use one way and the other the other way, eventually they have to learn and deal with both ways.
Brian Granger made similar arguments much better than I could here: https://youtu.be/OrpPDkZef5I

But I admit this new feature is nice, and decided to leave the decision on this to @HDembinski .
That is, unless @piti118 - if you chime in here, it would be very valuable.